app/com/linkedin/drelephant/tuning/hbt/MRApplicationData.java

```diff
@@ -0,0 +1,375 @@
+
+    return newSplitSize;
+  }
+
+  private long getNumberOfReducer(AppHeuristicResult yarnAppHeuristicResult) {
+    int numberoOfTasks = 0;
+    double averageTaskTimeInMinute = 0.0;
+    int newNumberOfReducer = 0;
+    for (AppHeuristicResultDetails appHeuristicResultDetails : yarnAppHeuristicResult.yarnAppHeuristicResultDetails) {
+      logger.info("Names " + appHeuristicResultDetails.name);
+      if (appHeuristicResultDetails.name.equals(AVERAGE_TASK_RUNTIME)) {
+        averageTaskTimeInMinute = getTimeInMinute(appHeuristicResultDetails.value);
+      }
+      if (appHeuristicResultDetails.name.equals(NUMBER_OF_TASK)) {
+        numberoOfTasks = Integer.parseInt(appHeuristicResultDetails.value);
+      }
+    }
+    addCounterData(new String[]{Reducer + " " + AVERAGE_TASK_RUNTIME, Reducer + " " + NUMBER_OF_TASK},
+        averageTaskTimeInMinute, numberoOfTasks * 1.0);
+    if (averageTaskTimeInMinute <= AVG_TASK_TIME_LOW_THRESHOLDS_FIRST) {
+      newNumberOfReducer = numberoOfTasks / SPLIT_SIZE_INCREASE_FIRST;
+    } else if (averageTaskTimeInMinute <= AVG_TASK_TIME_LOW_THRESHOLDS_SECOND) {
+      newNumberOfReducer = (int) (numberoOfTasks * SPLIT_SIZE_DECREASE);
+    } else if (averageTaskTimeInMinute >= AVG_TASK_TIME_HIGH_THRESHOLDS_FIRST) {
+      newNumberOfReducer = numberoOfTasks * SPLIT_SIZE_INCREASE_FIRST;
+    } else if (averageTaskTimeInMinute >= AVG_TASK_TIME_HIGH_THRESHOLDS_SECOND) {
+      newNumberOfReducer = (int) (numberoOfTasks * SPLIT_SIZE_INCREASE_SECOND);
+    }
+    logDebuggingStatement(" Reducer Average task time " + averageTaskTimeInMinute,
+        " Reducer Number of tasks " + numberoOfTasks * 1.0, " New number of reducer " + newNumberOfReducer);
+
+    return newNumberOfReducer;
+  }
+
+  private double getTimeInMinute(String value) {
+    value = value.replaceAll(" ", "");
+    String timeSplit[] = value.split("hr|min|sec");
+    double timeInMinutes = 0.0;
+    if (timeSplit.length == TimeUnit.sec.ordinal() + 1) {
+      timeInMinutes = timeInMinutes + Integer.parseInt(timeSplit[0]) * 60;
+      timeInMinutes = timeInMinutes + Integer.parseInt(timeSplit[1]);
+      timeInMinutes = timeInMinutes + Integer.parseInt(timeSplit[2]) * 1.0 / 60 * 1.0;
+    } else if (timeSplit.length == TimeUnit.min.ordinal() + 1) {
+      timeInMinutes = timeInMinutes + Integer.parseInt(timeSplit[0]);
+      timeInMinutes = timeInMinutes + Integer.parseInt(timeSplit[1]) * 1.0 / 60 * 1.0;
+    } else if (timeSplit.length == TimeUnit.hr.ordinal() + 1) {
+      timeInMinutes = timeInMinutes + Integer.parseInt(timeSplit[0]) * 1.0 / 60 * 1.0;
+    }
+    return timeInMinutes;
+  }
+
+  private void processForMemoryBuffer(AppHeuristicResult yarnAppHeuristicResult) {
+    float ratioOfDiskSpillsToOutputRecords = 0.0f;
+    int newBufferSize = 0;
+    float newSpillPercentage = 0.0f;
+    for (AppHeuristicResultDetails appHeuristicResultDetails : yarnAppHeuristicResult.yarnAppHeuristicResultDetails) {
```

Comment:
You can use getHeuristicsResultDetailsMap method from AppResult to get the map and then lookup in map to find all parameter values.